src/fmeval/transforms/util.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Dict, List
+from fmeval.model_runners.composers.composers import PromptComposer
+from fmeval.model_runners.model_runner import ModelRunner
+from fmeval.transforms.transform import Transform
+from fmeval.util import assert_condition
+
+
+def validate_existing_keys(record: Dict[str, Any], keys: List[str]) -> None:
+    """Validate that all expected keys are present in a record.
+
+    :param record: The record to be validated.
+    :param keys: The keys that are expected to be present in the record.
+    :raises: EvalAlgorithmInternalError if any validation fails.
+    """
+    for key in keys:
```

Comment:
This is a good point. I think that this util function shouldn't be the one doing the uniqueness validation. Instead, whoever is calling this function should do that validation themselves. Namely, since `keys` will almost always be the `input_keys` attribute of some `Transform` instance, we should update `Transform`'s initializer to enforce uniqueness in its `input_keys` and `output_keys`.